src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEEventSymbol.cs

```diff
@@ -96,11 +96,9 @@ private enum Flags : byte
 
                 const int targetSymbolCustomModifierCount = 0;
                 var typeSymbol = DynamicTypeDecoder.TransformType(originalEventType, targetSymbolCustomModifierCount, handle, moduleSymbol);
-                typeSymbol = TupleTypeDecoder.DecodeTupleTypesIfApplicable(typeSymbol, handle, moduleSymbol);
-
                 var type = TypeSymbolWithAnnotations.Create(typeSymbol);
                 type = NullableTypeDecoder.TransformOrEraseNullability(type, handle, moduleSymbol);
-
+                type = TupleTypeDecoder.DecodeTupleTypesIfApplicable(type, handle, moduleSymbol);
```

Comment:
>type = TupleTypeDecoder.DecodeTupleTypesIfApplicable(type, handle, moduleSymbol); [](start = 16, length = 81)

Is this move leads to an observable behavior change? If so, then probably DecodeTupleTypesIfApplicable is not preserving information that it is supposed to?

Reply:
It's not an observable change but it avoids converting between TupleTypeSymbol and NamedTypeSymbol unnecessarily.

---
In reply to: [194566546](https://github.com/dotnet/roslyn/pull/27589#discussion_r194566546) [](ancestors = 194566546)

Reply:
> It's not an observable change but it avoids converting between TupleTypeSymbol and NamedTypeSymbol unnecessarily.

Thanks for the explanation. It would be good to capture that in a comment so that this optimization wouldn't be undone by accident.

---
In reply to: [194617227](https://github.com/dotnet/roslyn/pull/27589#discussion_r194617227) [](ancestors = 194617227,194566546)